python-frontend/src/main/java/org/sonar/python/index/DescriptorUtils.java

```diff
@@ -53,10 +59,11 @@ private static ClassDescriptor classDescriptor(ClassSymbol classSymbol) {
       .withName(classSymbol.name())
       .withFullyQualifiedName(classSymbol.fullyQualifiedName())
       .withMembers(classSymbol.declaredMembers().stream().flatMap(s -> descriptors(s).stream()).collect(Collectors.toList()))
-      .withSuperClasses(classSymbol.superClasses().stream().map(Symbol::fullyQualifiedName).collect(Collectors.toList()))
+      .withSuperClasses(classSymbol.superClasses().stream().map(Symbol::fullyQualifiedName).filter(Objects::nonNull).collect(Collectors.toList()))
       .withDefinitionLocation(classSymbol.definitionLocation())
       .withHasMetaClass(((ClassSymbolImpl) classSymbol).hasMetaClass())
-      .withHasSuperClassWithoutDescriptor(((ClassSymbolImpl) classSymbol).hasSuperClassWithoutSymbol())
+      .withHasSuperClassWithoutDescriptor(((ClassSymbolImpl) classSymbol).hasSuperClassWithoutSymbol() ||
+        classSymbol.superClasses().stream().anyMatch(s -> s.fullyQualifiedName() == null))
```

Comment:
Yes, I hesitated a bit on that one.

Eventually I thought that a descriptor should be as accurate as it can be and that it's okay to infer some stuff from the original symbol when we create it: here we know from the null FQN that we won't be able to trace back to a descriptor nor a symbol, so I think saying "has super class without symbol" from the start is accurate and helps us avoid any edge case handling later on.

We allow symbols to have a sometimes "inconsistent" state because they're used in the local table for the analysis and we sometimes just need a placeholder. To me a descriptor is really a factory that we own to build symbols and I think it would be better if we never allow them to have such "inconsistent" state.

Reply:
Makes sense. I would anyway document this with a comment